src/esp/physics/bullet/BulletArticulatedObject.cpp

```diff
@@ -0,0 +1,808 @@
+    bWorld_->removeCollisionObject(linkCollider);
+    collisionObjToObjIds_->erase(linkCollider);
+    delete linkCollider;
+  }
+
+  // remove fixed base rigid body
+  if (bFixedObjectRigidBody_) {
+    bWorld_->removeRigidBody(bFixedObjectRigidBody_.get());
+    collisionObjToObjIds_->erase(bFixedObjectRigidBody_.get());
+    bFixedObjectRigidBody_ = nullptr;
+    bFixedObjectShape_ = nullptr;
+  }
+
+  // remove base collider
+  auto* baseCollider = btMultiBody_->getBaseCollider();
+  bWorld_->btCollisionWorld::removeCollisionObject(baseCollider);
+  collisionObjToObjIds_->erase(baseCollider);
+  delete baseCollider;
+
+  for (auto& jlIter : jointLimitConstraints) {
+    bWorld_->removeMultiBodyConstraint(jlIter.second.con);
+    delete jlIter.second.con;
+  }
+}
+
+void BulletArticulatedObject::initializeFromURDF(
+    URDFImporter& urdfImporter,
+    const Mn::Matrix4& worldTransform,
+    gfx::DrawableGroup* drawables,
+    scene::SceneNode* physicsNode,
+    bool fixedBase) {
+  Mn::Matrix4 rootTransformInWorldSpace{worldTransform};
+
+  BulletURDFImporter& u2b = *(static_cast<BulletURDFImporter*>(&urdfImporter));
+  u2b.setFixedBase(fixedBase);
+
+  auto urdfModel = u2b.getModel();
+
+  int flags = 0;
+
+  URDF2BulletCached cache;
+  u2b.InitURDF2BulletCache(cache, flags);
+
+  int urdfLinkIndex = u2b.getRootLinkIndex();
+  // int rootIndex = u2b.getRootLinkIndex();
+
+  // NOTE: recursive path only
+  u2b.ConvertURDF2BulletInternal(cache, urdfLinkIndex,
+                                 rootTransformInWorldSpace, bWorld_.get(),
+                                 flags, linkCompoundShapes_, linkChildShapes_);
+
+  if (cache.m_bulletMultiBody) {
+    btMultiBody* mb = cache.m_bulletMultiBody;
+    jointLimitConstraints = cache.m_jointLimitConstraints;
+
+    mb->setHasSelfCollision((flags & CUF_USE_SELF_COLLISION) !=
+                            0);  // NOTE: default no
+
+    mb->finalizeMultiDof();
+
+    btTransform localInertialFrameRoot =
+        cache.m_urdfLinkLocalInertialFrames[urdfLinkIndex];
+
+    {
+      mb->setBaseWorldTransform(btTransform(rootTransformInWorldSpace) *
+                                localInertialFrameRoot);
+    }
+    {
+      btAlignedObjectArray<btQuaternion> scratch_q;
+      btAlignedObjectArray<btVector3> scratch_m;
+      mb->forwardKinematics(scratch_q, scratch_m);
+      mb->updateCollisionObjectWorldTransforms(scratch_q, scratch_m);
+    }
+
+    btMultiBody_.reset(
+        mb);  // take ownership of the object in the URDFImporter cache
+    bWorld_->addMultiBody(btMultiBody_.get());
+    btMultiBody_->setCanSleep(true);
+
+    bFixedObjectShape_ = std::make_unique<btCompoundShape>();
+
+    // By convention, when fixed links in the URDF are assigned Noncollidable,
+    // we then insert corresponding fixed rigid bodies with group Static.
+    // Collisions with a fixed rigid body are cheaper than collisions with a
+    // fixed link, due to problems with multibody sleeping behavior.
+    {
```

Comment:
I refactored this into its own function and cleaned up the implementation a bit. I added a test and test asset. Note that I chose to have users set the `Static` collision group to trigger this functionality since I think it is more intuitive. 

We could probably also gate this feature on the `fixedBase` flag to prevent odd artifacts.